### ISSUE TITLE
ESI authorization not working?

### ISSUE THREAD

**Issue by rossos** [GoldenGnu/jeveassets#14]
I'm getting a 4xx-style ESI error when trying to add/edit/use ESI authorization.

I added ESI authorization some time ago, and it's apparently been working fine. Within the past few days, I got some errors that made me think I needed to do the authorization process again.

Going to Options -> Accounts, clicking the "Edit" button next to one of my characters, then clicking "Authorize", I am redirected in a browser to the following URL:

https://login.eveonline.com/Account/LogOn?ReturnUrl=/oauth/authorize?scope=esi-universe.read_structures.v1&response_type=code&state=jeveassets&redirect_uri=http://localhost:2221&client_id=

Notably, the client_id is blank. Once I authorize, I am given the following message by login.eveonline.com:

`{"error":"invalid_request","error_description":"Some parameters are either missing or invalid"}`

Pushing forward anyway and trying to fetch all new data ("Update -> Update"), I am (perhaps understandably) given ESI 400 error codes under "Accounts" and 401 error codes under "Structures".

**Comment by rossos** [comment]
I should have said that this is from version 4.1.4 (2017-08-18), according to Help -> About.

**Comment by GoldenGnu** [comment]
Thank you for your bug report.

I will release a new version later today, that fixes this problem.

Extra Info:
I had to re-upload 4.1.4 to the auto-update site, because of an error with
the release script.
When I did the re-upload I forgot to add the client information.

-Golden Gnu

**Comment by rossos** [comment]
After updating to 4.1.5, I can confirm that the ESI errors and missing client_id have both been resolved!

**Comment by GoldenGnu** [comment]
Thank you for reporting the bug.
I would never have found the bug myself. I never use the stable version.
I normal do a quick fire test before release, to make sure the most
important things are working.
But, I skipped it with the re-upload, as I already did the fire test when I
did the release the first time.
So, if nothing else came out of this, at least now I know to always do the
fire test, even if it's a re-upload.
